R/pkg/R/sparkR.R

```diff
@@ -369,6 +372,24 @@ sparkR.session <- function(
     overrideEnvs(sparkConfigMap, paramMap)
   }
 
+  # do not download if it is run in the sparkR shell
+  if (!nzchar(master) || isMasterLocal(master)) {
+    if (!isSparkRShell()) {
+      if (is.na(file.info(sparkHome)$isdir)) {
```

Comment:
I think this is from an outdated code - could you check your merge with master or edit this manually?